src/dotnet/Chat.Service/Controllers/UploadController.cs

```diff
@@ -1,26 +1,140 @@
-using ActualChat.Web.Internal;
+using ActualChat.Media;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Processing;
 
 namespace ActualChat.Chat.Controllers;
 
 [ApiController]
-public class UploadController : UploadControllerBase
+public class UploadController : ControllerBase
 {
-    private readonly IChats _chats;
+    private IContentSaver ContentSaver { get; }
+    private ILogger<UploadController> Log { get; }
+    private ICommander Commander { get; }
 
-    public UploadController(IChats chats) => _chats = chats;
+    public UploadController(
+        IContentSaver contentSaver,
+        ILogger<UploadController> log,
+        ICommander commander)
+    {
+        ContentSaver = contentSaver;
+        Log = log;
+        Commander = commander;
+    }
 
     [HttpPost, Route("api/chats/{chatId}/upload-picture")]
-    public Task<IActionResult> UploadPicture(ChatId chatId, CancellationToken cancellationToken)
+    public async Task<ActionResult<MediaContent>> UploadPicture(ChatId chatId, CancellationToken cancellationToken)
```

Comment:
Please ensure that method is backward compatible? if it's not - please create new one